===================================================================
RCS file: /var/lib/cvs/sources/freetype/freetype2/src/sfnt/ttcmap.c,v
retrieving revision 1.65
retrieving revision 1.66
diff -u -r1.65 -r1.66
--- freetype/freetype2/src/sfnt/ttcmap.c	2006/08/18 17:20:37	1.65
+++ freetype/freetype2/src/sfnt/ttcmap.c	2006/09/29 18:22:11	1.66
@@ -1094,14 +1094,19 @@
           /* search in segments before the current segment */
           for ( i = max ; i > 0; i-- )
           {
-            FT_UInt  prev_end;
+            FT_UInt   prev_end;
+            FT_Byte*  old_p;
 
 
-            p = cmap->data + 14 + ( i - 1 ) * 2;
+            old_p    = p;
+            p        = cmap->data + 14 + ( i - 1 ) * 2;
             prev_end = TT_PEEK_USHORT( p );
 
             if ( charcode > prev_end )
+            {
+              p = old_p;
               break;
+            }
 
             end    = prev_end;
             p     += 2 + num_segs2;
